README.md

```diff
@@ -154,31 +154,74 @@ It's great for API with request data format restrictions
   - decamelize
   - underscore
 
+#### `adapterOptions`
+Pass custom adapter options to handle them in `urlForCustomAction`. Required `AdpaterMixin` (instruction bellow).
 
-#### `promiseType`
-You can easily observe a returned model by changing promiseType to `array` or `object` according to what type of data
+#### `responseType`
+You can easily observe the returned model by changing `responseType` to `array` or `object` according to what type of data
 your server will return.
 
 When `array`:
 ```js
-model.customAction({}, { promiseType: 'array' }) // returns DS.PromiseArray
+model.customAction({}, { responseType: 'array' }) // returns DS.PromiseArray
 ```
 
 When `object`:
 ```js
-model.customAction({}, { promiseType: 'object' }) // returns DS.PromiseObject
+model.customAction({}, { responseType: 'object' }) // returns DS.PromiseObject
 ```
 
 When `null` (default):
 ```js
-model.customAction({}, { promiseType: null }) // returns Promise
+model.customAction({}, { responseType: null }) // returns Promise
 ```
 `null` is useful if you don't care about the response or just want to use `then` on the promise without using `binding` or display it in the template.
 
-#### `params`
+#### `queryParams`
 You can pass a query params for a request by passing an `{}` with properties, eg: `{ include: 'owner' }`
 ** Remember: Query params are not normalized! You have to pass it in the correct format. **
 
+### Adapter customization
+If you want to customize your request in your adapter please, implement our adapter mixin, eg:
+```js
+import JSONAPIAdapter from 'ember-data/adapters/json-api';
+import { AdapterMixin } from 'ember-custom-actions';
+
+export default JSONAPIAdapter.extend(AdapterMixin);
+```
+
+Then you can customize following methods:
+
+#### urlForCustomAction
+```js
+export default JSONAPIAdapter.extend(AdapterMixin, {
+  urlForCustomAction(modelName, id, snapshot, requestType, query) {
+    if (requestType === 'myPublishAction') {
+      return 'https://my-custom-api.com/publish'
+    }
+    
+    return this._super(...arguments);
+    let url = this._buildURL(modelName, id);
```

Comment:
Isn't that a dead code in here? I mean, the `return` above will return, and it won't get to that code.

Reply:
you right ;)